tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_symv_kk.hpp

```diff
@@ -0,0 +1,251 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SYMV_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SYMV_HPP_
+
+#include "signal_kokkos_impl_called.hpp"
+
+namespace KokkosKernelsSTD {
+
+namespace symv_impl{
+template <class size_type>
+constexpr bool static_extent_match(size_type extent1, size_type extent2)
+{
+  return extent1 == std::experimental::dynamic_extent ||
+         extent2 == std::experimental::dynamic_extent ||
+         extent1 == extent2;
+}
+} // end namespace symv_impl
+
+//
+// overwriting symmetric gemv: y = Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+	 class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class Triangle,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+requires (Layout_A::template mapping<std::experimental::extents<numRows_A, numCols_A>>::is_always_unique())
+void symmetric_matrix_vector_product(kokkos_exec<ExeSpace> /*kexe*/,
+				     std::experimental::mdspan<
+				       ElementType_A,
+				       std::experimental::extents<numRows_A, numCols_A>,
+				       Layout_A,
+				       std::experimental::default_accessor<ElementType_A>
+				     > A,
+				     Triangle /* tr */,
+				     std::experimental::mdspan<ElementType_x,
+				       std::experimental::extents<ext_x>,
+				       Layout_x,
+				       std::experimental::default_accessor<ElementType_x>
+				     > x,
+				     std::experimental::mdspan<ElementType_y,
+				       std::experimental::extents<ext_y>,
+				       Layout_y,
+				       std::experimental::default_accessor<ElementType_y>
+				     > y)
+{
+
+  // preconditions
+  if ( A.extent(0) != A.extent(1) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(0) != A.extent(1) ");
+  }
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+
+  symv_impl::static_extent_match(A.static_extent(0), A.static_extent(1));
+  symv_impl::static_extent_match(A.static_extent(1), x.static_extent(0));
+  symv_impl::static_extent_match(A.static_extent(0), x.static_extent(0));
+
+  auto A_view = Impl::mdspan_to_view(A);
```

Comment:
@fnrizzi @MikolajZuzek 

This reminds me that the spec doesn't say anything about not letting `A` be a transposed layout.  I think that's OK as far as the spec is concerned, because the algorithm will just access the set of `(i,j)` corresponding to the `Triangle` argument.  However, `mdspan_to_view` will need to handle this case.

Reply:
i opened this: https://github.com/kokkos/stdBLAS/issues/175

Reply:
one thought: when Kokkos view will be based on mdspan, I assume both layout, accessors will carry over right?

Reply:
@fnrizzi yup!  I haven't looked at @crtrott 's efforts on making Kokkos::View use mdspan, but I can imagine a couple different options for the accessor.  It might not necessarily be `default_accessor`.

Reply:
It's OK to address this later with a fix for #175.